tensorflow_probability/python/experimental/bijectors/highway_flow.py

```diff
@@ -1,27 +1,29 @@
 import numpy as np
-import tensorflow as tf
-
-import tensorflow_probability as tfp
-
-tfd = tfp.distributions
-tfb = tfp.bijectors
-tfe = tf.experimental
+import tensorflow.compat.v2 as tf
 
+from tensorflow_probability.python import bijectors as tfb
+from tensorflow_probability.python import util
+from tensorflow_probability.python.experimental.bijectors import scalar_function_with_inferred_inverse
 
 def build_highway_flow_layer(width, residual_fraction_initial_value=0.5, activation_fn=None):
     # FIXME: should everything be in float32 or float64?
```

Comment:
We typically try to infer the dtype that the user wants from the dtype of float-valued inputs. In this case you could write at the top:

```
residual_fraction_initial_value = tf.convert_to_tensor(residual_fraction_initial_value,
                                                       dtype_hint=tf.float32,
                                                       name='residual_fraction_initial_value')
dtype = residual_fraction_initial_value.dtype
```

This will use float32 by default, but adapt to float64 / float16 / etc. if the caller passes a Tensor value with explicit dtype (`convert_to_tensor` is a no-op if the input is already a Tensor). Then use `dtype` throughout when defining variables, e.g., 

`bias=tf.Variable(tf.random.normal([width], mean=0., stddev=0.01, dtype=dtype))`

etc.